src/main.rs

```diff
@@ -98,6 +105,120 @@ enum CallOrHash {
 	Hash([u8; 32]),
 }
 
+// All the info associated with a call in the forms you may need it in.
+struct CallInfo {
+	call: NetworkRuntimeCall,
```

Comment:
I'd use `Network` here instead since you never do anything with the data stored in `NetworkRuntimeCall`.